web/src/app/models/survey.model.ts

```diff
@@ -14,6 +14,15 @@
  * limitations under the License.
  */
 
+export enum OnboardingStatus {
```

Comment:
Nit: Rename to`SetupPhase`?

Reply:
Even better, can we remove this is favor of `State` := `INITIAL_SETUP` vs `null`? We can refer the state of the creation flow as follows:

1) Name survey: never - survey won't be saved until a name is specified.
2) Name job: when `jobs` is `null` or empty
3) Define tasks: when there is a job, but `tasks` is null or empty
4) "Who defines" LOI screen: new flag, for ex `dataCollectionType` - `PREDEFINED`, `AD_HOC`, `HYBRID`. Show this page when `dataCollectionType` is `null`
4) Define LOIs: when `dataCollectionType` is `STRUCTURED` or `HYBRID` but lois are empty.
5) Everything is set up, but `state == INITIAL_SETUP`.

Reply:
SG, made `SetupPhase` a enum for the component only and changed `OnboardingStatus` to `State`.